spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/AbstractConnectionFactory.java

```diff
@@ -367,11 +414,22 @@ public int getCloseTimeout() {
 	 */
 	public void setConnectionNameStrategy(ConnectionNameStrategy connectionNameStrategy) {
 		this.connectionNameStrategy = connectionNameStrategy;
+		if (this.publisherConnectionFactory != null) {
+			this.publisherConnectionFactory.setConnectionNameStrategy(
+					cf -> connectionNameStrategy + PUBLISHER_SUFFIX);
+		}
 	}
 
 	@Override
 	public void setBeanName(String name) {
 		this.beanName = name;
+		if (this.publisherConnectionFactory != null) {
+			this.publisherConnectionFactory.setBeanName(name + PUBLISHER_SUFFIX);
+		}
+	}
+
+	public boolean hasPublisherConnectionFactory() {
```

Comment:
I think we may have `PublisherConnectionFactoryAware` marker on this factory and use it from the producers to obtain this `publisherConnectionFactory` instance. See my comment in the `RabbitTemplate`

Reply:
I don't think we want to expose the sub-factory externally - otherwise they can configure them differently and the order in which they configure them would matter.